cg/meta/upload/gisaid/gisaid.py

```diff
@@ -0,0 +1,136 @@
+"""Interactions with the gisaid cli upload"""
+
+import logging
+from pathlib import Path
+from typing import List
+
+from cg.apps.housekeeper.hk import HousekeeperAPI
+from cg.models.cg_config import CGConfig
+from cg.store import models, Store
+from cg.utils import Process
+from .constants import HEADERS
+from .models import GisaidSample, FastaFile
+from housekeeper.store import models as hk_models
+import csv
+
+from cg.exc import HousekeeperVersionMissingError, FastaSequenceMissingError
+
+LOG = logging.getLogger(__name__)
+
+
+class GisaidAPI:
+    """Interface with Gisaid cli uppload"""
+
+    def __init__(self, config: CGConfig):
+        self.housekeeper_api: HousekeeperAPI = config.housekeeper_api
+        self.status_db: Store = config.status_db
+        self.gisaid_submitter: str = config.gisaid.submitter
+        self.gisaid_binary: str = config.gisaid.binary_path
+        self.process = Process(binary=self.gisaid_binary)
+
+    def get_fasta_sequence(self, fastq_path: str) -> str:
+        """Get header from fasta file. Assuming un zipped file"""
+
+        with open(fastq_path) as handle:
+            fasta_lines = handle.readlines()
+            try:
+                return fasta_lines[1].rstrip("\n")
+            except:
+                raise FastaSequenceMissingError
+
+    def get_gisaid_fasta_objects(self, gsaid_samples: List[GisaidSample]) -> List[FastaFile]:
+        """Fetch a fasta files form house keeper for batch upload to gisaid"""
+
+        fasta_objects = []
+        for sample in gsaid_samples:
+            hk_version: hk_models.Version = self.housekeeper_api.last_version(
```

Comment:
you can fetch housekeeper version once, since its the same bundle name for all files

Reply:
I mean, if you fetch it before the loop